### PR TITLE
Features/sim 18 academic strategy configuration

### PR DESCRIPTION
**Pull request by Zelalem-Fisseha**
i have completed the academic strategies with there respective tests and also with there respective blueprinters. adusted the way of overriding the crudaction using proc instead of blocks the test are written with rswag.

### CODE REVIEW

File: config/initializers/rswag_ui.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  # List the Swagger endpoints that you want to be documented through the
+  # swagger-ui. The first parameter is the path (absolute or relative to the UI
+  # host) to the corresponding endpoint and the second is a title that will be
+  # displayed in the document selector.
+  # NOTE: If you're using rspec-api to expose Swagger files
+  # (under openapi_root) as JSON or YAML endpoints, then the list below should
+  # correspond to the relative paths for those endpoints.
+
+  c.swagger_endpoint '/api-docs/v1/swagger.yaml', 'API V1 Docs'
```
Comment (Gutu-Dagmawi) [inline config/initializers/rswag_ui.rb]: change to:

``` ruby
c.openapi_endpoint '/api-docs/v1/swagger.yaml', 'API V1 Docs'
```
since it's going to be deprecated.

File: spec/factories/academic/engine/academic_timelines.rb
Comment (Gutu-Dagmawi) [inline spec/factories/academic/engine/academic_timelines.rb:1]: Why did you remove all the traits?

File: spec/factories/academic/engine/intakes.rb
Comment (Gutu-Dagmawi) [inline spec/factories/academic/engine/intakes.rb:1]: Same goes for this, won't we need the traits?

File: spec/factories/academic/engine/program_offerings.rb
Comment (Gutu-Dagmawi) [inline spec/factories/academic/engine/program_offerings.rb:1]: Again, why delete the traits? this implies you haven't tested the model comprehensively or that you've declared a lot of `let` blocks in the spec file.

File: spec/factories/academic/engine/programs.rb
Comment (Gutu-Dagmawi) [inline spec/factories/academic/engine/programs.rb:1]: Again with the traits..

File: app/models/academic/engine/program_offering.rb [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+
     with_options presence: true do
       validates :mode
-      validates :program
-      validates :intake
-      validates :academic_timeline
     end
 
-    validates :program, uniqueness: { scope: :intake }
-    validates :intake, uniqueness: { scope: :academic_timeline }
```
Comment (Gutu-Dagmawi) [inline app/models/academic/engine/program_offering.rb:16]: We should reinstate these validations, although by changing them up:

``` ruby
validates :program, uniqueness: { scope: { :intake, :mode, :academic_timeline, :campus } }
```
since we can't have a `program` belonging to a `program_offering` which has the same combination of `{ :intake, :mode, :academic_timeline, :campus }`

**Review by Gutu-Dagmawi — CHANGES_REQUESTED** [review]
I've left several comments regarding:

-  Test coverage and `factory_bot trait` usage
-  Validation issues
-  Using obsolete swagger config options

All else is correct, please resolve the issues ASAP.
> note: your specs only test the most basic functionalities and don't account for edge cases and multiple scenarios. Make sure to make your test suites a bit more comprehensive.

File: config/initializers/rswag_ui.rb (hunk 1 above)
Comment (Zelalem-Fisseha) [reply config/initializers/rswag_ui.rb]: you are wrong

File: spec/factories/academic/engine/academic_timelines.rb
Comment (Zelalem-Fisseha) [reply spec/factories/academic/engine/academic_timelines.rb:1]: will be included in the next pull request

File: spec/factories/academic/engine/intakes.rb
Comment (Zelalem-Fisseha) [reply spec/factories/academic/engine/intakes.rb:1]: they will be included in the next pull request

File: spec/factories/academic/engine/program_offerings.rb
Comment (Zelalem-Fisseha) [reply spec/factories/academic/engine/program_offerings.rb:1]: i have done the important improvements the others will be included in the next pull request

File: spec/integration/rintakes_spec.rb
Comment (Gutu-Dagmawi) [inline spec/integration/rintakes_spec.rb:1]: You should declare schemas for responses.

**Review by Gutu-Dagmawi — CHANGES_REQUESTED** [review]
